### PR TITLE
Fixes #27255 - katello-change-hostname cleans puppet certs

### PR DESCRIPTION
**Pull request by jturel**
Also fixes an additional bug when changing the hostname on a proxy
caused by giving the installer an invalid parameter

### CODE REVIEW

**Comment by tbrisker** [comment]
[test rpm]

**Comment by tbrisker** [comment]
[test rpm] (please disregard me, checking jenkins config)

**Comment by tbrisker** [comment]
[test rpm]

File: packages/katello/katello/hostname-change.rb [hunk 1]
```diff
@@ -167,6 +167,28 @@ def using_custom_certs?(scenario_answers)
...
+        'ca/inventory.txt',
+        'ca/infra_inventory.txt',
+        'ca/infra_serials',
+        'ca/serial',
+        'ca/root_key.pem',
+        'ca/ca_key.pem'
+      ]
+
+      files.each do |file|
+        run_cmd("rm -rf #{File.join(puppet_ssldir, file)}")
```
Comment (ekohl) [inline packages/katello/katello/hostname-change.rb]: Wouldn't it be cleaner to use native Ruby functions? Or at least quote the argument.

Comment (jturel) [reply packages/katello/katello/hostname-change.rb]: I think so, but run_cmd is consistent with everything else in the script and apparently it does some cleanup and handling of any errors

**Comment by tbrisker** [comment]
[test packaging]

File: packages/katello/katello/hostname-change.rb [hunk 2]
```diff
@@ -167,6 +167,28 @@ def using_custom_certs?(scenario_answers)
...
+        'ca/inventory.txt',
+        'ca/infra_inventory.txt',
+        'ca/infra_serials',
+        'ca/serial',
+        'ca/root_key.pem',
+        'ca/ca_key.pem'
+      ]
+
+      files.each do |file|
+        run_cmd("rm -f '#{File.join(puppet_ssldir, file)}'")
```
Comment (jturel) [inline packages/katello/katello/hostname-change.rb]: @ekohl quoted & changed to -f

**Comment by ekohl** [comment]
> This isn't related to this PR, but I see some JS errors on the products page, but I'm not sure if they were pre-existing before the katello change hostname run. Let me know if you've seen something similar!

These are fallout from the vendor js change.

File: packages/katello/katello/hostname-change.rb [hunk 3]
```diff
@@ -167,6 +167,28 @@ def using_custom_certs?(scenario_answers)
...
+      files = [
+        'ca/ca_crt.pem',
+        'ca/ca_crl.pem',
+        'ca/infra_crl.pem',
+        'certs/ca.pem',
+        'crl.pem',
+        'ca/ca_pub.pem',
+        'ca/inventory.txt',
+        'ca/infra_inventory.txt',
+        'ca/infra_serials',
```
Comment (ekohl) [inline packages/katello/katello/hostname-change.rb]: I don't see this on my Puppet 5 setup and I think you're using Puppet 6. Do we want to be safe and just remove the entire `puppet_ssldir`? AFAIK Puppet recreates it if needed and you don't need to track individual files.

File: packages/katello/katello/hostname-change.rb [hunk 4]
```diff
@@ -167,6 +167,12 @@ def using_custom_certs?(scenario_answers)
       scenario_answers["certs"]["server_cert_req"]
     end
 
+    def delete_puppet_certs
+      puppet_ssldir = @scenario_answers['puppet']['ssldir']
+
+      run_cmd("rm -rf '#{puppet_ssldir}'")
```
Comment (jturel) [inline packages/katello/katello/hostname-change.rb]: Updated this to remove the whole folder. It appears to work! I tested it on a nightly proxy. @johnpmitsch can you confirm it works on your katello nightly?

**Comment by evgeni** [comment]
[test rpm]

**Comment by jturel** [comment]
Looks like we're all good here, but I don't have merge access :)

**Comment by jturel** [comment]
> Both are issues in 1.22 as well so candidates for a cherry pick.
> 
> Edit: I'll leave it the pick request up to someone else :)

I flagged the issue for Katello 3.12.2 so we'll include it there

**Pull request merged**